internal/status/statuses.go

```diff
@@ -67,8 +67,12 @@ type GatewayClassStatus struct {
 	ObservedGeneration int64
 }
 
-// buildStatuses builds statuses from a Graph.
-func buildStatuses(graph *graph.Graph) Statuses {
+type NginxReloadResult struct {
+	Error error
+}
+
+// BuildStatuses builds statuses from a Graph.
+func BuildStatuses(graph *graph.Graph, nginxReloadRes NginxReloadResult) Statuses {
```

Comment:
because of the refactor, do we really need to expose the Statuses type anymore?  I wonder if it became just an implementation detail of the package

Reply:
Exporting this is allowing us to unit test it at the moment. This is where blackbox unit testing is painful.